src/Server.cpp

```diff
@@ -46,85 +87,127 @@ void Server::runServer() {
   }
 }
 
-void shrink_to_fit(std::vector<struct pollfd>& vec) {
-  std::vector<struct pollfd>(vec).swap(vec);
-}
-
 void Server::closeServer() {
-  for (size_t i = 0; i < _clients.size(); ++i) {
-    std::cout << RED << "Client <" << _clients[i].getFd();
-    std::cout << "> Disconnected" << RESET << std::endl;
-    close(_clients[i].getFd());
+  // Fermer tous les clients
+  for (clientsMap::iterator it = _clients.begin(); it != _clients.end(); it++) {
+    closeClient(it->second.getFd());
   }
   _clients.clear();
+
+  // Fermer le socket principal
   if (_socketFd != -1) {
-    std::cout << RED << "Server <" << _socketFd;
-    std::cout << "> Disconnected" << RESET << std::endl;
+    std::cout << RED "Server <" << _socketFd << "> Disconnected" << std::endl;
     close(_socketFd);
     _socketFd = -1;
   }
-  for (size_t i = 0; i < _pollFds.size(); ++i) {
-    if (_pollFds[i].fd != -1) {
-      close(_pollFds[i].fd);
-    }
-  }
   _pollFds.clear();
-  shrink_to_fit(_pollFds);
 }
-
 void Server::signalHandler(int signal) {
   if (signal == SIGINT || signal == SIGQUIT) {
-    std::cout << std::endl << "Signal Received!" << std::endl;
-    Server::_signal = true;
+    _signal = true;
+    std::cout << std::endl << WHITE "Signal Received!" << std::endl;
   }
 }
 
-void Server::clearClient(int fd) {
-  for (size_t i = 0; i < _pollFds.size(); i++) {
-    if (_pollFds[i].fd == fd) {
-      _pollFds.erase(_pollFds.begin() + i);
-      break;
-    }
+/* Clients Management */
+
+void Server::handleClientMessage(int fd) {
+  char buffer[1024] = {0};
+  std::memset(buffer, 0, sizeof(buffer));
+  int valread = recv(fd, buffer, sizeof(buffer), 0);
+
+  switch (valread) {
+  case -1:
+    std::cerr << RED "Error while receiving message" << std::endl;
+  case 0:
+    std::cout << "Client " << fd << " disconnected" << std::endl;
+    clearClient(fd);
+    return;
   }
-  for (size_t i = 0; i < _clients.size(); i++) {
-    if (_clients[i].getFd() == fd) {
-      _clients.erase(_clients.begin() + i);
-      break;
+
+  std::string message(buffer, valread);
+  std::cout << "Received message from client " << fd << ": " << message
+            << std::endl;
+  std::istringstream iss(message);
+  std::string command;
+  iss >> command;
+
+  if (command == "JOIN") {
```

Comment:
We would be able to replace this part by `handle command` function. (I didn't test the new version yet.)
```
  if (command == "JOIN") {
    std::string channelName;
    iss >> channelName;
    if (_channels.find(channelName) == _channels.end()) {
      _channels[channelName] = Channel(channelName);
    }
    _channels[channelName].acceptClientInTheChannel(_clients[fd]);
  } else {
    sendToAllClients(message);
  }
```

```
void Server::handleCommand(const std::string &command, int fd) {
  static_cast<void>(fd);
  if (command.empty()) {
    return;
  } else if (command == "JOIN") {
     std::string channelName;
    iss >> channelName;
    if (_channels.find(channelName) == _channels.end()) {
      _channels[channelName] = Channel(channelName);
    }
    _channels[channelName].acceptClientInTheChannel(_clients[fd]);
  } else if (command == "KICK") {
    // Exclure un client du canal
  } else if (command == "INVITE") {
    // Notice
  } else if (command == "TOPIC") {
    // Changer le sujet du canal
  } else if (command == "MODE") {
    // Changer le sujet du canal
  } else if (command == "LIST") {
    // Lister les canaux
  } else if (command == "NOTICE") {
    // Notice}
  } else if (command == "NICK") {
    // change nickname
  } else if (command == "PRIVMSG") {
    // Envoyer un message privé
  } else if (command == "QUIT") {
    // Déconnecter le client
  } else if (command == "PING") {
    // client.sendNumericReply(1, "PONG");
  } else {
    // Temporary code
    sendToAllClients(message);
  }
}
```